nck/readers/sa360_reader.py

```diff
@@ -0,0 +1,131 @@
+# GNU Lesser General Public License v3.0 only
+# Copyright (C) 2020 Artefact
+# licence-information@artefact.com
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+import csv
+import click
+
+from io import StringIO
+
+from nck.commands.command import processor
+from nck.readers.reader import Reader
+from nck.utils.args import extract_args
+from nck.streams.normalized_json_stream import NormalizedJSONStream
+from nck.clients.sa360_client import SA360Client
+from nck.helpers.sa360_helper import REPORT_TYPES
+
+DATEFORMAT = "%Y-%m-%d"
+ENCODING = "utf-8"
+
+
+@click.command(name="read_sa360")
+@click.option("--sa360-access-token", default=None)
+@click.option("--sa360-client-id", required=True)
+@click.option("--sa360-client-secret", required=True)
+@click.option("--sa360-refresh-token", required=True)
+@click.option("--sa360-agency-id", required=True)
+@click.option(
+    "--sa360-advertiser-id",
+    "sa360_advertiser_ids",
+    multiple=True,
+    help="If empty, all advertisers from agency will be requested",
+)
```

Comment:
You can't have only the Advertiser ID but not the Agency ID (just like Google Ads, where sometimes you only have Account IDs, but no Manager Account ID)

Reply:
Nope, in SA360, you have agencies, several advertisers in an agency with several accounts per advertiser
![sa360](https://user-images.githubusercontent.com/15572717/78795575-73610100-79b5-11ea-9ef8-0a79c322ea65.png)